go/pkg/reporter.go

```diff
@@ -75,45 +80,47 @@ func MergeTags(tags0, tags1 []string) []string {
 	return mergedTags
 }
 
-func (reporter *HumbugReporter) Publish(report Report) error {
+func (reporter *HumbugReporter) Publish(report Report) {
 	defer func() {
 		// TODO(zomglings): For now, we only recover here so panicked Publish calls do not have an
 		// effect on any program using the Humbug library. In the future, there should be an option
 		// of writing log messages if an environment variable (e.g. HUMBUG_LOG_LEVEL) has been
 		// set appropriately.
 		recover()
 	}()
-	var err error
-	userHasConsented := reporter.consent.Check()
+	userHasConsented := reporter.Consent.Check()
 	if userHasConsented {
-		context := spire.EntryContext{
-			ContextType: "humbug",
-			ContextID:   reporter.sessionID,
-		}
 		tags := MergeTags(report.Tags, reporter.Tags())
-		_, err = reporter.bugoutClient.Spire.CreateEntry(reporter.bugoutAccessToken, reporter.bugoutJournalID, report.Title, report.Content, tags, context)
+		entriesRoute := fmt.Sprintf("%s/humbug/reports", reporter.BaseUrl)
+		requestBody := reportRequest{
+			Title:   report.Title,
+			Content: report.Content,
+			Tags:    tags,
+		}
+		requestBuffer := new(bytes.Buffer)
+		json.NewEncoder(requestBuffer).Encode(requestBody)
+
+		request, _ := http.NewRequest("POST", entriesRoute, requestBuffer)
+
+		client := &http.Client{}
+		request.Header.Add("Content-Type", "application/json")
+		request.Header.Add("Accept", "application/json")
+		request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", reporter.ReporterToken))
+	
+		client.Do(request)
+
 	}
-	return err
 }
 
-func CreateHumbugReporter(consent Consent, clientID string, sessionID string, bugoutAccessToken string, bugoutJournalID string) (*HumbugReporter, error) {
-	reporter := HumbugReporter{
-		consent:           consent,
-		clientID:          clientID,
-		sessionID:         sessionID,
-		bugoutAccessToken: bugoutAccessToken,
-		bugoutJournalID:   bugoutJournalID,
-	}
+func CreateHumbugReporter(reporter HumbugReporter ) (*HumbugReporter, error) {
```

Comment:
This is inconvenient for users.

Signature should be:
```go
func CreateHumbugReporter(consent Consent, clientID string, sessionID string, reporterToken string) (*HumbugReporter, error) {
    ...
}
```

If someone needs to change `baseURL`, the pattern should be like this:
```go
reporter := CreateHumbugReporter(...)
reporter.baseURL = newBaseURL
```

Reply:
Default `baseURL` on reporter (created by `CreateHumbugReporter`) should be `https://spire.bugout.dev`

Reply:
Or add `HumbugReporter.SetBaseURL(baseURL string)` method.

Reply:
```go
func (reporter *HumbugReporter) SetBaseURL(baseURL string) {
    reporter.baseURL = baseURL
}
```